packages/terra-button-group/src/ButtonGroupButton.jsx

```diff
@@ -63,6 +67,17 @@ class ButtonGroupButton extends React.Component {
     }
   }
 
+  handleClick(event) {
```

Comment:
Do we need handleClick event..? if it is declared just to set `shouldShowFocus` to true then I see we are handling `shouldShowFocus` in keydown event as well which will set `shouldShowFocus` to true tab key press I think this should enough to hold the focus styles on selected button. 

having handleClick() here is making same value 'true' to set twice for `shouldShowFocus`.

Reply:
@pranav300  correct me if i am wrong. @supreethmr I think in the case of first switching to another window or tabbing (which will make this shouldShowFocus to true). so once we come back to the button and click it. the variable shouldShowFocus will still be true. in that case we need handle click.

Reply:
We can set `shouldShowFocus` to false in handleBlur to handle this scenario. 
For example :
```jsx
 handleOnBlur(event) {
    if (document.activeElement === event.currentTarget && this.state.focused) {
      this.shouldShowFocus = true;
    } else {
      this.shouldShowFocus = false;
      this.setState({ focused: false });
    }

    if (this.props.onBlur) {
      this.props.onBlur(event);
    }
  }
``` 
Since blur will be called on both scenarios like switching to new window or when we click outside / other button. handling `this.shouldShowFocus` in blur event would fix this issue. 

Above code will hold back the focus styles on button in blur event if it is previously focused ( on leaving the window ) else removes focus styles from button. So we might don't need handle click event to set `this.shouldShowFocus` to false it can be handled in handleBlur like shown above.

Reply:
Removed onClick event here https://github.com/cerner/terra-core/pull/2933/commits/372398dc76e2c279355e476dee13fa8e5e587fc2